lib.nix

```diff
@@ -209,7 +209,11 @@ rec
                 pushd $out/$member > /dev/null
                 mkdir -p src
                 # Avoid accidentally pulling `std` for no-std crates.
-                echo '#![no_std]' >src/lib.rs
+                cat <<EOF >src/lib.rs
+                #![no_std]
```

Comment:
```suggestion
                #![no_std]
                # If you compile with crate-type=cdylib, and also use no_std, 
                # the compilation will fail because no panic handler has been
                # provided (the bit that unwinds the stack trace, usually provided 
                # by the standard library). This is just a dummy stub to get
                # compilation to succeed. 
                # See https://doc.rust-lang.org/nomicon/panic-handler.html for more details.
```